core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/ChunkBuffer.java

```diff
@@ -21,45 +21,45 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 import java.util.concurrent.atomic.AtomicLong;
 import software.amazon.awssdk.annotations.SdkInternalApi;
 import software.amazon.awssdk.utils.Logger;
-import software.amazon.awssdk.utils.Validate;
 import software.amazon.awssdk.utils.builder.SdkBuilder;
 
 /**
- * Class that will buffer incoming BufferBytes of totalBytes length to chunks of bufferSize*
+ * Class that will buffer incoming BufferBytes to chunks of bufferSize.
+ * If totalBytes is not provided, i.e. content-length is unknown, {@link #getBufferedData()} should be used in the Subscriber's
+ * {@code onComplete()} to check for a final chunk that is smaller than the chunk size, and send if present.
  */
 @SdkInternalApi
 public final class ChunkBuffer {
     private static final Logger log = Logger.loggerFor(ChunkBuffer.class);
-    private final AtomicLong transferredBytes;
+    private AtomicLong transferredBytes;
```

Comment:
why are we removing `final`?

Reply:
must have removed in earlier iterations, added back